### PR TITLE
Update devDependencies.

### PR DESCRIPTION
**Pull request by sttk**
I've updated devDependencies in package.json and have made tests succeed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
   ],
   "scripts": {
     "lint": "eslint . && jscs index.js bin/ lib/ test/",
-    "prepublish": "marked-man --name gulp docs/CLI.md > gulp.1",
+    "prepare": "marked-man --name gulp docs/CLI.md > gulp.1",
+    "prepublishOnly": "npm run prepare",
```
Comment (phated) [inline package.json]: I still use npm@2 so this won't work for me.

File: .gitignore [hunk 2]
```diff
@@ -14,3 +14,4 @@ test/out-fixtures/*
 test/output/
 test/watch-*.txt
 gulp.1
+package-lock.json
```
Comment (phated) [inline .gitignore]: Makes sense but does npm pick up `.npmrc` files from the repo?  Might be better to include an npmrc that stops the package-lock creation.  Also, belongs in a separate commit.

File: package.json [hunk 3]
```diff
@@ -51,20 +52,20 @@
...
+    "babel-preset-es2015": "^6.24.1",
     "babel-register": "^6.5.1",
-    "coveralls": "^2.7.0",
+    "coveralls": "^2.13.1",
     "eslint": "^1.7.3",
-    "eslint-config-gulp": "^2.0.0",
+    "eslint-config-gulp": "^2.1.0",
     "expect": "^1.20.2",
-    "gulp": "gulpjs/gulp#4.0",
+    "gulp": "^4.0.0",
```
Comment (phated) [inline package.json]: I think we should only update this in the dep update PR.

File: test/fixtures/packages/invalid-package.json [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
-    "node": ">= 0.9"
-  },
-  "licenses": [
-    {
-      "type": "MIT",
-      "url": "https://raw.githubusercontent.com/gulpjs/gulp/master/LICENSE"
-    }
-  ]
-}
+{
```
Comment (phated) [inline test/fixtures/packages/invalid-package.json]: These files were causing problems on my machine due to weird line ending stuff.  I had to tell git to ignore the changes to make them go away.

File: package.json (hunk 1 above)
Comment (sttk) [reply package.json]: Sorry. `prepare` and `prepublishOnly` was introduced in npm@4. I'll put back this.

File: .gitignore (hunk 2 above)
Comment (sttk) [reply .gitignore]: I got it. I'll commit .npmrc separately.

**Comment by sttk** [comment]
@phated I've removed unnecessary modifications and have separated into commits of updating dev dependencies and disabling package-lock.json.

File: package.json [hunk 5]
```diff
@@ -57,13 +57,13 @@
     "eslint": "^1.7.3",
     "eslint-config-gulp": "^2.0.0",
     "expect": "^1.20.2",
-    "gulp": "gulpjs/gulp#4.0",
+    "gulp": "^4.0.0",
     "gulp-test-tools": "^0.6.1",
-    "jscs": "^2.3.5",
+    "jscs": "^3.0.7",
```
Comment (phated) [inline package.json]: I thought jscs changed a bunch of stuff in 3.0 that made our config file invalid?

File: package.json [hunk 6]
```diff
@@ -57,13 +57,13 @@
...
     "jscs-preset-gulp": "^1.0.0",
-    "marked-man": "^0.1.3",
+    "marked-man": "^0.2.1",
```
Comment (phated) [inline package.json]: What changed in this module?

File: package.json (hunk 5 above)
Comment (sttk) [reply package.json]: [Release note of jscs 3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) wrote:
> We didn't rename/remove any rules. ...

So I think there are no problem to use 3.0.

File: package.json (hunk 6 above)
Comment (sttk) [reply package.json]: Just I think newer module is better. Doesn't `^0.1.3` represents the range `>=0.1.3 <0.2.0`?

**Comment by phated** [comment]
@sttk can you also bump minor versions for any dependencies that are above 1.0?  Ref #147

**Comment by sttk** [comment]
I've update more dependency versions,  fancy-log and so on.

**Comment by phated** [comment]
@sttk I cherry-picked your commits into master and squashed some stuff (needed to deal with line endings of those fixtures on my new computer) - I'll get this published with the exitCode thing.

**Comment by sttk** [comment]
@phated I got it. Thanks!
